src/main/kotlin/io/sc3/peripherals/posters/printer/PosterPrinterScreenHandler.kt

```diff
@@ -0,0 +1,100 @@
+package io.sc3.peripherals.posters.printer
+
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.entity.player.PlayerInventory
+import net.minecraft.inventory.Inventory
+import net.minecraft.inventory.SimpleInventory
+import net.minecraft.item.ItemStack
+import net.minecraft.screen.ArrayPropertyDelegate
+import net.minecraft.screen.PropertyDelegate
+import net.minecraft.screen.ScreenHandler
+import net.minecraft.screen.slot.Slot
+import io.sc3.peripherals.Registration.ModItems
+import io.sc3.peripherals.Registration.ModScreens.posterPrinter
+import io.sc3.peripherals.util.PropertyDelegateGetter
+import io.sc3.peripherals.util.ValidatingSlot
+import net.minecraft.item.Items
+
+const val PAPER_SLOT = 0
+const val INK_SLOT = 1
+const val OUTPUT_SLOT = 2
+
+const val INV_SIZE = 3
+
+class PosterPrinterScreenHandler(
+  syncId: Int,
+  playerInv: PlayerInventory,
+  private val inv: Inventory,
+  propertyDelegate: PropertyDelegate
+) : ScreenHandler(posterPrinter, syncId) {
+  private val paperSlot: Slot
+  private val inkSlot: Slot
+  private val outputSlot: Slot
+
+  val ink by PropertyDelegateGetter(propertyDelegate, 1)
+  val printProgress by PropertyDelegateGetter(propertyDelegate, 2)
+  val maxPrintProgress by PropertyDelegateGetter(propertyDelegate, 3)
+
+  constructor(syncId: Int, playerInv: PlayerInventory) :
+    this(syncId, playerInv, SimpleInventory(3), ArrayPropertyDelegate(4))
+
+  init {
+    checkSize(inv, 3)
+
+    // TODO: Validating slots
```

Comment:
This TODO can be removed, I removed it on the 3D printer one too - the validating slots are the lines below! :D